src/util/string.h

```diff
@@ -37,3 +39,31 @@ inline QString coloredLinkString(
             QStringLiteral(";\" href=\"") + baseUrl + extUrl +
             QStringLiteral("\">") + text + QStringLiteral("</a>");
 }
+
+// Check if the extension from the file filter was added to the file base name.
+// Otherwise add it manually.
+// Works around https://bugreports.qt.io/browse/QTBUG-27186
+inline QString filePathWithSelectedExtension(const QString& fileLocationDlg,
+        const QString& fileFilter) {
+    if (fileLocationDlg.isEmpty()) {
+        return QString();
+    }
+    QString fileLocation = fileLocationDlg;
+    if (fileFilter.isEmpty()) {
+        return fileLocation;
+    }
+
+    // Extract 'ext' from QFileDialog display string 'Funky type (*.ext)'
+    QString ext;
+    const QRegularExpression extRgxp("(\\*\\.)(.*?)(\\))");
```

Comment:
make static, put into anonymous namespace.
I'm also struggling to understand the regex at all.

Reply:
okay.
the regex (group 2) catches anything in between `*.` and `)`. idk if there's a better way do achieve this.
For escaping we need double backslashes, otherwise it fails to compile with "invalid escape sequence".

Reply:
In regards to the invalid escape sequence, afaik C++ has raw string literals (`R"(\\\\)"` is `"\x5C\x5C\x5C\x5C"` `\x5C` ASCII repr of backslash)

I propose the following regex:
```suggestion
    const QRegularExpression extRgxp(R"(\(\*\.(.*)\)$)");
```
it removes the unnecessary capturing groups, unnecessary quantifiers and is consistent with escaping reserved identifiers (since you forgot that for some braces). It also adds the restriction that the `(*.ext)` is at the end of the line (so it can't match that pattern as part of the filename)

Reply:
> C++ has raw string literals

This is neat, the $ eol assertion, too. Thanks!